### ISSUE TITLE
Changing parent of taxon with rank causes api link in rank

### ISSUE THREAD

**Issue by melton-jason** [specify/specify7#2637]
**Describe the bug**
While a taxon record has a Rank, editing the parent of the taxon causes the Rank field to be populated with an api rank. 
Note: the correct rank associated with the newly selected parent can still be selected from the pick list. 

New Taxon:

https://user-images.githubusercontent.com/64045831/207375036-6d0a4a92-4018-40c7-9a85-072510efa396.mp4

Existing Taxon:

https://user-images.githubusercontent.com/64045831/207375752-0053fb11-0225-4ce6-b1c2-cf8f3b11277d.mp4

**Desktop:**
 - OS: Windows 11
 - Browser: Chrome
 - Specify 7 Version: testability [7.8.0]

**Comment by melton-jason** [comment]
This issue occurs with Geography records as well

**Comment by maxxxxxdlp** [comment]
Actually, this probably occurs on edge too.

Here is why:
Parent is Animalia. You selected a Phylum to be current rank
You changed parent to be much lower in the tree. Now Phylum is an invalid tree rank as it is above the new parent

Don't know if this would happen often in the real world
And if so, how do we want to handle this? I.e, reset the rank when parent changes AND new rank is invalid?

**Comment by melton-jason** [comment]
If need be, we can discuss ways to handle this, but I agree with the solution you proposed. Reset the rank when parent changes (and new rank is invalid).
This will probably rarely occur in production databases (if ever), so regardless of a solution it can likely wait for a future point release.

**Comment by tlammer** [comment]
I agree it can be in a future release though I think when the new parent makes the old rank invalid, you should null out the rank instead of assigning a new one.

**Comment by bronwyncombs** [comment]
Changing the parent on a taxon record to a synonymized taxon rank breaks the Taxonomic Ranks field and crashes on save

![Screenshot 2023-08-28 at 11 52 00 AM](https://github.com/specify/specify7/assets/135047322/df89919c-dcfb-464e-bf1d-989dcc1c3b93)

[Specify 7 Crash Report - 2023-08-28T17_20_56.586Z (1).txt](https://github.com/specify/specify7/files/12457773/Specify.7.Crash.Report.-.2023-08-28T17_20_56.586Z.1.txt)

 - macOS, Chrome, `v7.13`, `fwri_6_12_23`

**Comment by bronwyncombs** [comment]
In Litho Strat tree as of`v7.9-dev`: [Relevant comment](https://github.com/specify/specify7/issues/3577#issuecomment-1710958924)

Also reported by Heather Cole at AAFC-AAC:

> Currently in our Geography Tree, we have England, Scotland, Wales and Northern Ireland at the “country” level. None of them have any nodes at the state/province level (labelled ‘primary division’ in our tree). We want to ‘move’ all four nodes down a level, and put them as primary divisions in a country called “United Kingdom of Great Britain and Northern Ireland” (they all have secondary nodes already, but nothing at the primary level).
> I created the new country, and I thought I would be able to use move/merge to move the existing nodes in, but it seems like there may be a few ‘tree rules’ in place that are stopping me from doing that.

On `v7.9.1-arm`

**Comment by bronwyncombs** [comment]
Also in Firefox -- edge, v7.9.1
1. Create new tree node
2. Carry forward w/o parent name
3. Add new parent name of different rank in tree
4. See error

<img width="1904" alt="Screenshot 2023-12-08 at 9 46 42 AM" src="https://github.com/specify/specify7/assets/135047322/5b5fdde2-b187-42b4-9a48-d9a94bb90637">

**Comment by grantfitzsimmons** [comment]
More likely related information here: https://github.com/specify/specify7/issues/3577
